src/gluonts/model/rotbaum/_preprocess.py

```diff
@@ -275,10 +279,21 @@ def __init__(
         use_feat_static_real=False,
         use_feat_dynamic_real=False,
         use_feat_dynamic_cat=False,
-        cardinality="auto",
-        one_hot_encode: bool = True,  # should improve accuracy but will slow down model
+        cardinality: Union[List[int], str] = "auto",
+        one_hot_encode: bool = True,  # Should improve accuracy but will slow down model
         **kwargs
     ):
+
+        if one_hot_encode:
+            assert cardinality != "ignore" or (
+                isinstance(cardinality, List)
+                and all(c > 0 for c in cardinality)
+            ), "You should set `one_hot_encode=True` if and only if cardinality is a valid list or not ignored"
```

Comment:
This doesn't look too right to me.

I understand `one_hot_encode` as an option which can be enabled, but it doesn't require that there are actually categorical features.